### PR TITLE
Aerospike exporter - new integration

### PR DESCRIPTION
**Pull request by pkosanam**
JIRA Ticket: https://issues.newrelic.com/browse/NR-67664

Description:
We are trying to integrate a new quickstart named "[Aerospike](https://aerospike.com/)"
Hence, I have created this PR to make sure that users can use nri-aerospike to install the aerospike-prometheus-exporter binary

Aerospike prometheus exporter: https://github.com/aerospike/aerospike-prometheus-exporter

### CODE REVIEW

**Review by carlossscastro — CHANGES_REQUESTED** [review]
Nice work so far!
We're almost there.
A few changes to make the E2E pass + a couple of other changes needed.
Some of the changes required were introduced by us when we implemented multi OS and Architecture support in the exporter packages.

**Comment by pkosanam** [comment]
Thank you @carlossscastro , I have updated all the files as guided.

On executing make-test-e2e-aerospike \ ... \ ... on my local environment, I am seeing the below error but was able to capture metrics after few mins on one.newrelic.com

Is it common or is there any other configuration check which I am missing ?

Observing error while executing make-test-e2e command:
image

Observing metrics on one.newrelic.com after few minutes:
image

Thank you. 😃
cc: @mdumpati

File: exporters/aerospike/ape.toml.tmpl [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+# Supports below formats,
+# 1. Credential directly                                                      - "<credential>"
+# 2. Credential via file                                                      - "file:<file-that-contains-credential>"
+# 3. Credential via environment variable                                      - "env:<environment-variable-that-contains-credential>"
+# 4. Credential via environment variable containing base64 encoded credential - "env-b64:<environment-variable-that-contains-base64-encoded-credential>"
+# 5. Credential in base64 encoded form                                        - "b64:<base64-encoded-credential>"
+# Applicable to 'user' and 'password' configurations.
+
+# database user
+user = ""
```
[the remark this answers is not in the document]
Comment (pkosanam) [reply exporters/aerospike/ape.toml.tmpl:94]: I have added all the required fields on the /exporters/aerospike/aerospike.config.yml file and update the ape.toml.tmpl file accordingly.

Kindly update me if there is anything that I can change here. 😃

File: exporters/aerospike/exporter.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# Tag of the exporter to checkout
+exporter_tag:
+# Commit of the exporter to checkout (used if tag property is empty)
+exporter_commit: 612e842ea3e56fdee42589dc886e5efbc5d2e639
+# Changelog to add to the new release
+exporter_changelog: https://github.com/aerospike/aerospike-prometheus-exporter/blob/master/CHANGELOG.md
+# Enable packages for Linux
+package_linux: true
+# (Optional) Exporter external config files (multiple files can be added separated by commas. i.e. file1,file2)
+exporter_config_files: ape.toml
```
[the remark this answers is not in the document]
Comment (pkosanam) [reply exporters/aerospike/exporter.yml:20]: added "package_linux_goarchs: amd64,arm64"

File: exporters/aerospike/build-exporter-linux.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+if [[ -z "${EXPORTER_TAG}" ]]
+then
+    git -c advice.detachedHead=false checkout "${EXPORTER_COMMIT}"
+else
+    git checkout "${EXPORTER_TAG}" -d
+fi
+
+# ###############################################################
+#  Build exporter
+make
```
[the remark this answers is not in the document]
Comment (pkosanam) [reply exporters/aerospike/build-exporter-linux.sh:35]: added.. 👍

File: exporters/aerospike/e2e/e2e_spec.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+          exporter_port: 9145
+          label_type: development
+          label_source: aerospike
+          exporter_files_path: /tmp
+
+    tests:
+      nrqls:
+        - query: "FROM Metric SELECT latest(aerospike_node_stats_client_connections)"
+      entities:
+        - type: "AEROSPIKE_NODE"
```
Comment (carlossscastro) [inline exporters/aerospike/e2e/e2e_spec.yml:28]: According to the entity-definition (https://github.com/newrelic/entity-definitions/blob/main/definitions/ext-aerospike/definition.yml#L2) we only have an Entity `AEROSPIKE`, so you need to adjust the test to the correct entity name and remove `AEROSPIKE_NODE` and `AEROSPIKE_NAMESPACE`

File: exporters/aerospike/e2e/e2e_spec.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+          data_type: "Metric"
+          metric_name: "aerospike_node_stats_client_connections"
+        - type: "AEROSPIKE_NAMESPACE"
+          data_type: "Metric"
+          metric_name: "aerospike_namespace_available_bin_names"
+      metrics:
+        - source: "aerospike.yml"
+          except_entities:
+            - AEROSPIKE_NAMESPACE
```
Comment (carlossscastro) [inline exporters/aerospike/e2e/e2e_spec.yml:37]: This line can be removed also as we don't have `AEROSPIKE_NAMESPACE` entity

**Review by carlossscastro — CHANGES_REQUESTED** [review]

**Comment by carlossscastro** [comment]
Regarding the pipeline error, could you refetch your fork and rebase. It should get rid of the E2E error.

File: exporters/aerospike/e2e/e2e_spec.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+          except_metrics:
+            - aerospike_namespace_available_bin_names
+            - aerospike_namespace_allow_ttl_without_nsup
+            - aerospike_namespace_appeals_records_exonerated
+            - aerospike_namespace_appeals_rx_active
+            - aerospike_namespace_appeals_tx_active
+            - aerospike_namespace_appeals_tx_remaining
+            - aerospike_namespace_available_bin_names
+            - aerospike_namespace_cache_read_pct
```
Comment (carlossscastro) [inline exporters/aerospike/e2e/e2e_spec.yml:46]: All these metrics seem to be created by the test environment so they shouldn't be excluded

**Review by carlossscastro — CHANGES_REQUESTED** [review]

**Comment by pkosanam** [comment]
I am closing this PR as I have created another PR for aerospike.
new PR: https://github.com/newrelic/newrelic-prometheus-exporters-packages/pull/163

**Pull request closed without merging**